lib/plurimath/latex/parser.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_relative "parse"
+require_relative "constants"
+require_relative "transform"
+module Plurimath
+  class Latex
+    class Parser
+      attr_accessor :text
+
+      def initialize(text)
+        @text = text.gsub(" ", "").gsub("\n", "")
+      end
+
+      def parse
+        tree_t = Plurimath::Latex::Parse.new.parse(text)
+        formula = Plurimath::Latex::Transform.new.apply(tree_t)
+        formula = [formula] unless formula.is_a?(Array) || formula.nil?
+        return if formula.nil?
```

Comment:
I think `nil` should not be returned, always return the formula object even if it is empty.